*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 17:25:16 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /usr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uWSGI http bound on 127.0.0.1:5000  // 启动地址 fd 7
uwsgi socket 0 bound to TCP address 127.0.0.1:45369 (port auto-assigned) fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
Python main interpreter initialized at 0x55b84623ab60
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 17743)
spawned uWSGI worker 1 (pid: 17744, cores: 10)
spawned uWSGI worker 2 (pid: 17745, cores: 10)
spawned uWSGI http 1 (pid: 17746)
--- no python application found, check your startup logs for errors ---
[pid: 17745|app: -1|req: -1/1] 127.0.0.1 () {50 vars in 991 bytes} [Thu Nov 28 17:25:39 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 2)
--- no python application found, check your startup logs for errors ---
[pid: 17745|app: -1|req: -1/2] 127.0.0.1 () {48 vars in 923 bytes} [Thu Nov 28 17:25:39 2024] GET /favicon.ico => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
DAMN ! worker 1 (pid: 17744) died, killed by signal 9 :( trying respawn ...
Respawned uWSGI worker 1 (new pid: 21239)
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 20:18:57 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
Python main interpreter initialized at 0x55f219a604f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 17553)
spawned uWSGI worker 1 (pid: 17554, cores: 10)
spawned uWSGI worker 2 (pid: 17555, cores: 10)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 20:19:20 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
Python main interpreter initialized at 0x55f5eb5014f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 17733)
spawned uWSGI worker 1 (pid: 17734, cores: 10)
spawned uWSGI worker 2 (pid: 17735, cores: 10)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 20:30:06 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
Python main interpreter initialized at 0x562e1ce5c4f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 19711)
spawned uWSGI worker 1 (pid: 19712, cores: 10)
spawned uWSGI worker 2 (pid: 19713, cores: 10)
invalid request block size: 21573 (max 4096)...skip
--- no python application found, check your startup logs for errors ---
[pid: 19713|app: -1|req: -1/1] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 21:07:58 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 1)
--- no python application found, check your startup logs for errors ---
[pid: 19713|app: -1|req: -1/2] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 21:11:54 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 4)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 21:12:29 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
Python main interpreter initialized at 0x55b0e8c224f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 27948)
spawned uWSGI worker 1 (pid: 27949, cores: 10)
spawned uWSGI worker 2 (pid: 27950, cores: 10)
--- no python application found, check your startup logs for errors ---
[pid: 27949|app: -1|req: -1/1] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 21:12:38 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 2)
--- no python application found, check your startup logs for errors ---
[pid: 27950|app: -1|req: -1/2] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 22:34:03 2024] GET / => generated 21 bytes in 2 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 2)
--- no python application found, check your startup logs for errors ---
[pid: 27949|app: -1|req: -1/3] 127.0.0.1 () {54 vars in 1028 bytes} [Thu Nov 28 22:34:18 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
--- no python application found, check your startup logs for errors ---
[pid: 27949|app: -1|req: -1/4] 127.0.0.1 () {52 vars in 959 bytes} [Thu Nov 28 22:34:19 2024] GET /favicon.ico => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 1)
--- no python application found, check your startup logs for errors ---
[pid: 27949|app: -1|req: -1/5] 127.0.0.1 () {56 vars in 1059 bytes} [Thu Nov 28 22:35:05 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 3)
--- no python application found, check your startup logs for errors ---
[pid: 27949|app: -1|req: -1/6] 127.0.0.1 () {52 vars in 959 bytes} [Thu Nov 28 22:35:05 2024] GET /favicon.ico => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 5)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 22:54:40 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x55a4ec38d330
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
failed to open python file manage.py  // 项目的启动文件
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 14141)
spawned uWSGI worker 1 (pid: 14142, cores: 10)
spawned uWSGI worker 2 (pid: 14143, cores: 10)
--- no python application found, check your startup logs for errors ---
[pid: 14142|app: -1|req: -1/1] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 22:55:03 2024] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 1)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 22:59:21 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x5623adcb73d0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (GET, HEAD, OPTIONS) -> static>,
 <Rule '/user/adduser/' (OPTIONS, POST) -> user.add_user>,
 <Rule '/user/' (GET, HEAD, OPTIONS) -> user.users>,
 <Rule '/' (GET, HEAD, OPTIONS) -> home.home>])
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x5623adcb73d0 pid: 15076 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 15076)
spawned uWSGI worker 1 (pid: 15081, cores: 10)
spawned uWSGI worker 2 (pid: 15082, cores: 10)
[pid: 15082|app: 0|req: 1/1] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 22:59:27 2024] GET / => generated 30 bytes in 4 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 0)
[pid: 15082|app: 0|req: 2/2] 127.0.0.1 () {32 vars in 347 bytes} [Thu Nov 28 23:00:19 2024] GET /user => generated 239 bytes in 1 msecs (HTTP/1.1 308) 3 headers in 134 bytes (1 switches on core 2)
[pid: 15082|app: 0|req: 3/3] 127.0.0.1 () {32 vars in 349 bytes} [Thu Nov 28 23:00:31 2024] GET /users => generated 207 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 3)
[pid: 15082|app: 0|req: 4/4] 127.0.0.1 () {32 vars in 347 bytes} [Thu Nov 28 23:00:33 2024] GET /user => generated 239 bytes in 0 msecs (HTTP/1.1 308) 3 headers in 134 bytes (1 switches on core 1)
[pid: 15081|app: 0|req: 1/5] 127.0.0.1 () {32 vars in 347 bytes} [Thu Nov 28 23:01:15 2024] GET /user => generated 239 bytes in 6 msecs (HTTP/1.1 308) 3 headers in 134 bytes (1 switches on core 0)
[pid: 15082|app: 0|req: 5/6] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 23:01:18 2024] GET / => generated 30 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 6)
[pid: 15082|app: 0|req: 6/7] 127.0.0.1 () {32 vars in 339 bytes} [Thu Nov 28 23:01:22 2024] GET / => generated 30 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 7)
hello world
2024-11-28 23:01:35,744 - app - ERROR - Exception on /user/ [GET]
Traceback (most recent call last):
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/home/power/flask_try/./app/controller/users.py", line 21, in users
    app.logger.info("hello")
AttributeError: module 'flask.app' has no attribute 'logger'
[pid: 15082|app: 0|req: 7/8] 127.0.0.1 () {32 vars in 349 bytes} [Thu Nov 28 23:01:35 2024] GET /user/ => generated 265 bytes in 6 msecs (HTTP/1.1 500) 2 headers in 99 bytes (1 switches on core 9)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 23:02:27 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x556d415843d0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (GET, OPTIONS, HEAD) -> static>,
 <Rule '/user/adduser/' (POST, OPTIONS) -> user.add_user>,
 <Rule '/user/' (GET, OPTIONS, HEAD) -> user.users>,
 <Rule '/' (GET, OPTIONS, HEAD) -> home.home>])
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x556d415843d0 pid: 15864 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 15864)
spawned uWSGI worker 1 (pid: 15866, cores: 10)
spawned uWSGI worker 2 (pid: 15867, cores: 10)
hello world
2024-11-28 23:02:39,179 - app - ERROR - Exception on /user/ [GET]
Traceback (most recent call last):
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "/root/.local/share/virtualenvs/flask_try-1EnXE9dr/lib/python3.8/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
  File "/home/power/flask_try/./app/controller/users.py", line 21, in users
    app.logger.info("hello")
AttributeError: module 'flask.app' has no attribute 'logger'
[pid: 15867|app: 0|req: 1/1] 127.0.0.1 () {32 vars in 349 bytes} [Thu Nov 28 23:02:39 2024] GET /user/ => generated 265 bytes in 10 msecs (HTTP/1.1 500) 2 headers in 99 bytes (1 switches on core 0)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Nov 28 23:27:55 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /usr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31255
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x5610938c9570
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (HEAD, GET, OPTIONS) -> static>,
 <Rule '/user/adduser/' (POST, OPTIONS) -> user.add_user>,
 <Rule '/user/' (HEAD, GET, OPTIONS) -> user.users>,
 <Rule '/' (HEAD, GET, OPTIONS) -> home.home>])
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x5610938c9570 pid: 17194 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 17194)
spawned uWSGI worker 1 (pid: 17202, cores: 10)
spawned uWSGI worker 2 (pid: 17203, cores: 10)
hello world
[pid: 17203|app: 0|req: 1/1] 127.0.0.1 () {32 vars in 349 bytes} [Thu Nov 28 23:28:10 2024] GET /user/ => generated 11 bytes in 6 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 0)
hello world
[pid: 17202|app: 0|req: 1/2] 127.0.0.1 () {32 vars in 349 bytes} [Thu Nov 28 23:28:30 2024] GET /user/ => generated 11 bytes in 5 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 0)
hello world
[pid: 17203|app: 0|req: 2/3] 127.0.0.1 () {54 vars in 1038 bytes} [Thu Nov 28 23:38:57 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 2)
[pid: 17203|app: 0|req: 3/4] 127.0.0.1 () {52 vars in 964 bytes} [Thu Nov 28 23:38:57 2024] GET /favicon.ico => generated 207 bytes in 1 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 4)
hello world
[pid: 17203|app: 0|req: 4/5] 172.17.240.1 () {38 vars in 745 bytes} [Thu Nov 28 23:39:43 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 1)
[pid: 17203|app: 0|req: 5/6] 172.17.240.1 () {38 vars in 697 bytes} [Thu Nov 28 23:39:43 2024] GET /favicon.ico => generated 207 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 5)
hello world
[pid: 17203|app: 0|req: 6/7] 172.17.240.1 () {40 vars in 776 bytes} [Thu Nov 28 23:39:46 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 3)
hello world
[pid: 17203|app: 0|req: 7/8] 172.17.240.1 () {40 vars in 776 bytes} [Thu Nov 28 23:39:47 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 6)
hello world
[pid: 17203|app: 0|req: 8/9] 127.0.0.1 () {32 vars in 349 bytes} [Fri Nov 29 08:51:09 2024] GET /user/ => generated 11 bytes in 1 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 7)
hello world
[pid: 17203|app: 0|req: 9/10] 172.17.249.58 () {32 vars in 357 bytes} [Fri Nov 29 08:51:38 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 8)
hello world
[pid: 17203|app: 0|req: 10/11] 172.17.249.58 () {32 vars in 357 bytes} [Fri Nov 29 08:53:32 2024] GET /user/ => generated 11 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 9)
[pid: 17203|app: 0|req: 11/12] 172.17.249.58 () {32 vars in 347 bytes} [Fri Nov 29 08:53:37 2024] GET / => generated 30 bytes in 1 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 0)
hello world
[pid: 17203|app: 0|req: 12/13] 127.0.0.1 () {32 vars in 348 bytes} [Fri Nov 29 09:07:14 2024] GET /user/ => generated 11 bytes in 1 msecs (HTTP/1.1 200) 2 headers in 79 bytes (1 switches on core 2)
[pid: 17203|app: 0|req: 13/14] 127.0.0.1 () {32 vars in 338 bytes} [Fri Nov 29 09:07:20 2024] GET / => generated 30 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 4)
[pid: 17203|app: 0|req: 14/15] 172.17.249.58 () {32 vars in 347 bytes} [Fri Nov 29 09:53:40 2024] GET / => generated 30 bytes in 1 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 1)
*** Starting uWSGI 2.0.28 (64bit) on [Thu Dec  5 15:49:45 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /usr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31203
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x555ce38c73a0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (HEAD, GET, OPTIONS) -> static>,
 <Rule '/user/adduser/' (POST, OPTIONS) -> user.add_user>,
 <Rule '/user/' (HEAD, GET, OPTIONS) -> user.users>,
 <Rule '/' (HEAD, GET, OPTIONS) -> home.home>])
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x555ce38c73a0 pid: 7738 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 7738)
spawned uWSGI worker 1 (pid: 7740, cores: 10)
spawned uWSGI worker 2 (pid: 7741, cores: 10)
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
invalid request block size: 21573 (max 4096)...skip
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 0 seconds
worker 2 buried after 0 seconds
goodbye to uWSGI.
*** Starting uWSGI 2.0.28 (64bit) on [Thu Dec  5 16:05:41 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /usr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31203
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x561ab79793a0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (GET, HEAD, OPTIONS) -> static>,
 <Rule '/user/adduser/' (POST, OPTIONS) -> user.add_user>,
 <Rule '/user/' (GET, HEAD, OPTIONS) -> user.users>,
 <Rule '/' (GET, HEAD, OPTIONS) -> home.home>])
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x561ab79793a0 pid: 8001 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 8001)
spawned uWSGI worker 1 (pid: 8003, cores: 10)
spawned uWSGI worker 2 (pid: 8007, cores: 10)
*** Starting uWSGI 2.0.28 (64bit) on [Thu Dec  5 16:17:34 2024] ***
compiled with version: 9.4.0 on 28 November 2024 08:02:56
os: Linux-5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024
nodename: LUO
machine: x86_64
clock source: unix
detected number of CPU cores: 24
current working directory: /home/power/flask_try
detected binary path: /root/.local/share/virtualenvs/flask_try-1EnXE9dr/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 31203
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:8000 fd 6
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.8.10 (default, Nov  7 2024, 13:10:47)  [GCC 9.4.0]
PEP 405 virtualenv detected: /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Set PythonHome to /root/.local/share/virtualenvs/flask_try-1EnXE9dr
Python main interpreter initialized at 0x55c1ed2e34f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 500592 bytes (488 KB) for 20 cores
*** Operational MODE: preforking+threaded ***
app created
Map([<Rule '/static/<filename>' (HEAD, GET, OPTIONS) -> static>,
 <Rule '/user/adduser/' (POST, OPTIONS) -> user.add_user>,
 <Rule '/user/' (HEAD, GET, OPTIONS) -> user.users>,
 <Rule '/' (HEAD, GET, OPTIONS) -> home.home>])
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x55c1ed2e34f0 pid: 2852 (default app)
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 2852)
spawned uWSGI worker 1 (pid: 2854, cores: 10)
spawned uWSGI worker 2 (pid: 2855, cores: 10)
[pid: 2854|app: 0|req: 1/1] 127.0.0.1 () {56 vars in 1059 bytes} [Thu Dec  5 16:17:42 2024] GET / => generated 30 bytes in 9 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 1)
[pid: 2854|app: 0|req: 2/2] 127.0.0.1 () {56 vars in 1059 bytes} [Thu Dec  5 16:18:24 2024] GET / => generated 30 bytes in 0 msecs (HTTP/1.1 200) 2 headers in 71 bytes (1 switches on core 4)
SIGINT/SIGTERM received...killing workers...
worker 1 buried after 0 seconds
worker 2 buried after 0 seconds
goodbye to uWSGI.
